### PR TITLE
fix: updated ci actions

### PR DESCRIPTION
**Pull request by sabind**
## Description

Updated deploy scripts.

## Release Checklist

- [ ] This code has unit tests
- [ ] I have a plan to verify that these changes are working as expected after deploy
- [ ] I have a plan for how to revert, rollback, or disable these changes if things are not working as expected

## Security Checklist

This PR creates, modifies, or deletes:

- [ ] API routes: API routes, parameters, or user authorization
- [ ] Authentication: Authentication mechanism
- [ ] Credentials: Server side credentials, or secrets in configuration / source code
- [ ] Cryptography: Encryption, hashing, certificates, signatures, random numbers, etc.
- [ ] User data: personal information handling, logs, error messages, etc.

### CODE REVIEW

File: .firebaserc [hunk 1]
```diff
@@ -28,21 +31,27 @@
...
+        ],
+        "site": [
+          "reach4help-staging"
         ]
       }
     },
     "reach4help": {
       "hosting": {
         "web-client": [
-          "reach4help-web-client"
```
Comment (sabind) [inline .firebaserc:37]: When the project moved we lost this name to the old `reach4help-xxxx` project :(

**Comment by s0** [comment]
Hmm, this looks good, but i don't know whether this branching + environment mapping makes sense. Notably, i don't think auto deploying to dev is a good idea, isn't that environment supposed to be for us to run and test things while we're developing them, that makes it a bit harder to do if it's automatically deployed to by master right?

And regarding the other branches, why not just have them named `staging` and `production` vs `release` and `prerelease` to simplify it?

**Comment by sabind** [comment]
@s0 I could see using `staging` and `production` branches vs. using the tags and releases features. I don't have the permissions to create the new protected branches, but if you'd like to make them I can update the scripts to leverage them.

**Comment by s0** [comment]
> I don't have the permissions to create the new protected branches, but if you'd like to make them I can update the scripts to leverage them.

That'll be because they already exist :)

Any thoughts on not auto-deploying master to dev too?

**Comment by sabind** [comment]
We can push development -> dev, staging -> staging, prod -> prod and then master is the just final place where we archive the code after successful releases?

**Comment by s0** [comment]
Uhh... I'm still convinced that `dev` should be for manual deployments only, we need an environment where we can test things during development (before PR), and during review (after PR), if not Dev, then we need to create another environment specifically for this purpose.

**Comment by sabind** [comment]
If we keep dev for manual deploys then we won't have anywhere to work on things as a team. If you need ephemeral deploys for testing PRs then users can simply deploy to their own Firestore env and add the URL in their PR. Some deploys to dev may cause data that isn't backwards compatible so deploying to that environment should be regulated until after PRs are approved.

I recommend we go with this flow where we have one branch per env and if we need pull request builds then we can work on that as a precursor to dev (which also follows the branches).

**Comment by s0** [comment]
> If we keep dev for manual deploys then we won't have anywhere to work on things as a team. If you need ephemeral deploys for testing PRs then users can simply deploy to their own Firestore env and add the URL in their PR. Some deploys to dev may cause data that isn't backwards compatible so deploying to that environment should be regulated until after PRs are approved.
> 
> I recommend we go with this flow where we have one branch per env and if we need pull request builds then we can work on that as a precursor to dev (which also follows the branches).

Okay cool that's reasonable. Users making their own projects, or us creating them when needed isn't crazy.

In that case, i'm going to be a pain again and say we should use `master` -> `dev`, not `development` -> `dev`, the development branch's sole purpose is to accept PRs that come from forks, as unfortunately github actions don't run on those PRs, and we need those checks to be required. :)

**Comment by s0** [comment]
I'll revert your PR back to `f975847` and merge :)

**Comment by sabind** [comment]
Ok this makes sense - PRs -> Dev then we merge Dev -> Master to deploy, then we have staging and prod for the remaining envs.

**Pull request merged**